app/Database/InProgCourseQueries.hs

```diff
@@ -0,0 +1,70 @@
+--| UberGoal: go from T.Text -> [Entity Lecture] / [Entity Tutorial] without multiple DB queries.
+
+import Data.Tuple.Select as Tup -- selecting first, second and third elements of tuples (fall, spring, year)
+
+-- | Step 1: single query for lecture or tutorial. Produces list of all 
+-- | lectures/tutorials respectively, whcih are not separated by session
+
+-- | Queries db for all matching records with lecture or tutorial code of this course
+lectureQuery :: T.Text <- [Entity Lecture]
```

Comment:
This type isn't right. `selectList` is a database action, which has type `MonadIO => ReaderT SqlBackend m a`, where `a` is the "returned" type. In this case, this would be `[Entity Lecture]`, for a better type of `ReaderT SqlBackend m a`.

The `T.Text` as the argument type of the function is correct. So the full type is something like `T.Text -> ReaderT ...`, plus the `MonadIO` constraint.